spring-cloud-stream/src/test/java/org/springframework/cloud/stream/function/SourceToFunctionsSupportTests.java

```diff
@@ -0,0 +1,229 @@
+						.web(WebApplicationType.NONE)
+						.run("--spring.cloud.stream.function.name=toUpperCase", "--spring.jmx.enabled=false")){
+
+			OutputDestination target = context.getBean(OutputDestination.class);
+			assertThat(target.receive(1000).getPayload()).isEqualTo("HELLO FUNCTION".getBytes(StandardCharsets.UTF_8));
+		}
+	}
+
+	@Test
+	public void testComposedFunctionIsAppliedToExistingMessageSource() {
+		try(ConfigurableApplicationContext context =
+				new SpringApplicationBuilder(
+						TestChannelBinderConfiguration.getCompleteConfiguration(FunctionsConfiguration.class))
+						.web(WebApplicationType.NONE)
+						.run("--spring.cloud.stream.function.name=toUpperCase|concatWithSelf",
+								"--spring.jmx.enabled=false")) {
+			OutputDestination target = context.getBean(OutputDestination.class);
+			assertThat(target.receive(1000).getPayload())
+					.isEqualTo("HELLO FUNCTION:HELLO FUNCTION".getBytes(StandardCharsets.UTF_8));
+		}
+	}
+
+	@Test
+	public void testMessageSourceIsCreatedFromProvidedSupplier() {
+		try(ConfigurableApplicationContext context =
+				new SpringApplicationBuilder(
+						TestChannelBinderConfiguration.getCompleteConfiguration(SupplierConfiguration.class))
+						.web(WebApplicationType.NONE)
+						.run("--spring.cloud.stream.function.name=number",
+								"--spring.jmx.enabled=false")){
+
+			OutputDestination target = context.getBean(OutputDestination.class);
+			assertThat(target.receive(10000).getPayload())
+				.isEqualTo("1".getBytes(StandardCharsets.UTF_8));
+			assertThat(target.receive(10000).getPayload())
+				.isEqualTo("2".getBytes(StandardCharsets.UTF_8));
+			assertThat(target.receive(10000).getPayload())
+				.isEqualTo("3".getBytes(StandardCharsets.UTF_8));
+			//etc
+		}
+	}
+
+	@Test
+	public void testMessageSourceIsCreatedFromProvidedSupplierComposedWithSingleFunction() {
+		try(ConfigurableApplicationContext context =
+				new SpringApplicationBuilder(
+						TestChannelBinderConfiguration.getCompleteConfiguration(SupplierConfiguration.class))
+						.web(WebApplicationType.NONE)
+						.run("--spring.cloud.stream.function.name=number|concatWithSelf",
+								"--spring.jmx.enabled=false")){
+
+			OutputDestination target = context.getBean(OutputDestination.class);
+			assertThat(target.receive(10000).getPayload())
+				.isEqualTo("11".getBytes(StandardCharsets.UTF_8));
+			assertThat(target.receive(10000).getPayload())
+				.isEqualTo("22".getBytes(StandardCharsets.UTF_8));
+			assertThat(target.receive(10000).getPayload())
+				.isEqualTo("33".getBytes(StandardCharsets.UTF_8));
+			//etc
+		}
+	}
+
+	@Test
+	public void testMessageSourceIsCreatedFromProvidedSupplierComposedWithMultipleFunctions() {
+		try(ConfigurableApplicationContext context =
+				new SpringApplicationBuilder(
+						TestChannelBinderConfiguration.getCompleteConfiguration(SupplierConfiguration.class))
+						.web(WebApplicationType.NONE)
```

Comment:
Typo: need single `m` in the beginning of `multiplyByTwo`.

Fixing during merge...